File: BHAVRP_Java/BHAVRPv2.0/BHAVRP2.0/src/cujae/inf/ic/om/assignment/classical/urgency/IUrgencyWithMU.java
package cujae.inf.ic.om.assignment.classical.urgency;

import java.util.ArrayList;

import cujae.inf.ic.om.matrix.NumericMatrix;
import cujae.inf.ic.om.problem.input.Customer;

public interface IUrgencyWithMU {
	
	/*Retorna un listado con las urgencias de los clientes del listado entrado por parámetro*/
	ArrayList<Double> getListUrgencies(ArrayList<Customer> listCustomersToAssign, ArrayList<ArrayList<Integer>> listIDDepots, NumericMatrix urgencyMatrix, int muIDDepot);
	
	/*Método encargado de obtener la urgencia*/
	double getUrgency(int idCustomer, ArrayList<Integer> listIDDepots, NumericMatrix urgencyMatrix, int muIDDepot);

}

File: BHAVRP_Python/cujae/inf/citi/om/assignment/classical/urgency/i_urgency_with_mu.py
import numpy as np
from typing import List
from abc import ABC, abstractmethod
from ....problem.input.customer import Customer

class IUrgencyWithMU(ABC):
    
    @abstractmethod
    def get_list_urgencies(
        self,
        list_customers: List[Customer],
        list_id_depots: List[List[int]],
        urgency_matrix: np.ndarray,
        mu_id_depot: int
    ) -> List[float]:
        pass
    
    @abstractmethod
    def get_urgency(
        self,
        id_customer: int,
        list_id_depots: List[int],
        urgency_matrix: np.ndarray,
        mu_id_depot: int
    ) -> float:
        pass